README.md

```diff
@@ -1,16 +1,144 @@
+Chances are you did. Don't worry if it didn't leap out for you. Our process will
+help you spot the issue. Let's analyze our process for validating our
+pseudocode, the 6<sup>th</sup> step in the Flatiron Programming Process.
 
-## SWBAT 2
+## Identify a Process to Validate Pseudocode
+
+Here's the pseudocode again:
+
+```text
+Procedure DivideBaguetteEvenly(baguette, n):
+  baguette_length = measure(baguette)
+  even_length = baguette_length / n
+  collection = []
+
+  while baguette_length > even_length:
+    piece, rest = cut_bread(baguette, even_length)
+    collection.add(piece)
+
+    baguette = rest
+    baguette_length = measure(baguette)
+
+  even_pieces = collection
+  return even_pieces
+```
+
+We're going to walk through our pseudocode using a _state table_.
+
+### Introducing State Tables
+
+State tables are tables that keep track of the inputs, outputs, local
+variables, and conditions in a bounded or fenced-off "bloc" of code or
+pseudocode.
+
+Because the blocs are fenced off, sometimes we have to provide input that would
+be available if the bloc were part of a larger program. It's okay to do this
+because we'll assume that other code does its job properly.
+
+So for `DivideBaguetteEvenly`, we can create the following table:
+
+|Variable|State|State|State|State|State|State|State|State|State|
+|--------|-----|-----|-----|-----|-----|-----|-----|-----|-----|
+|Input: `baguette`|||||||||||
+|Input: _n_|||||||||||
+|Output: even_pieces|||||||||||
+|Local: `baguette_length`|||||||||||
+|Local: `even_length`|||||||||||
+|Local: `collection`|||||||||||
+|Cond: `baguette_length > even_length`|||||||||
+|Local: `piece`|||||||||||
+|Local: `rest`|||||||||||
+
+We'll "plug in" some data for `DivideBaguetteEvenly` use and then we'll step
+through, line by line and update the "state" or the "values" of the variables
+as we work through the _Procedure_'s function. We'll check our output against our
+expectations and make sure our logic is right.
+
+### Filling in the State Table
+
+Lets play through our _Procedure_ like so:
+
+`DivideBaguetteEvenly( 60cm baguette, 3 people)`
+
+We would expect that:
+
+* Each person gets a 20cm length piece
+* The baguette is length 0 at the end (we used the whole thing)
+
+Work through the code, filling out the table:
+
+|Variable|State|State|State|State|State|State|State|State|State|
+|--------|-----|-----|-----|-----|-----|-----|-----|-----|-----|
+|Input: `baguette`|60cm baguette|40cm baguette|20cm baguette||||||||
+|Input: _n_|3||||||||||
+|Output: even_pieces|[20cm piece,20cm piece]||||||||||
+|Local: `baguette_length`|60|40|20||||||||
+|Local: `even_length`|20||||||||||
+|Local: `collection`|[]|[20cm piece,20cm piece]|||||||||
+|Cond: `baguette_length > even_length`|T|T|**F**|
+|Local: `piece`|20cm piece|20cm piece|||||||||
+|Local: `rest`|40cm baguette|20cm baguette|||||||||
+
+**Output**: 2 pieces. This is not what we predicted and is also incorrect.
```

Comment:
I think we need to walk through this in more detail, move through the code and populate the first column as you go to show how we arrived at these values. I know taht would be a lot more...

Maybe a line by row sort of comparison? Show the blank table, step through the code and populate the table at the same time. This is just a suggestion, though, whatever you think makes sense